firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -235,31 +235,41 @@ private void persistFid(String fid) throws FirebaseInstallationsException {
   }
 
   /**
-   * Registers the FID with FIS servers if FID is in UNREGISTERED state.
+   * Registers the FID with FIS servers if FID is in UNREGISTERED state. Also, refreshes auth token
+   * if expired.
    *
    * <p>Updates FID registration status to PENDING to avoid multiple network calls to FIS Servers.
    */
   private Task<String> registerFidIfNecessary(
       PersistedFidEntry persistedFidEntry, AwaitListener listener) {
     String fid = persistedFidEntry.getFirebaseInstallationId();
 
-    // Check if the fid is unregistered
-    if (persistedFidEntry.getRegistrationStatus() == RegistrationStatus.UNREGISTERED) {
-      updatePersistedFidWithPendingStatus(fid);
-      executeFidRegistration(persistedFidEntry, listener);
-    } else {
-      updateAwaitListenerIfRegisteredFid(persistedFidEntry, listener);
+    if (persistedFidEntry.getRegistrationStatus() != RegistrationStatus.UNREGISTERED
+        && !isAuthTokenExpired(persistedFidEntry)) {
+      return updateAwaitListenerIfRegisteredFid(persistedFidEntry, listener);
     }
 
+    updatePersistedFidWithPendingStatus(persistedFidEntry);
+    executeFISCalls(persistedFidEntry, listener);
     return Tasks.forResult(fid);
   }
 
-  private void updateAwaitListenerIfRegisteredFid(
+  private void executeFISCalls(PersistedFidEntry persistedFidEntry, AwaitListener listener) {
+
+    if (!isAuthTokenExpired(persistedFidEntry)) {
+      executeFidRegistration(persistedFidEntry, listener);
+      return;
+    }
+    refreshAuthToken(listener);
+  }
+
+  private Task<String> updateAwaitListenerIfRegisteredFid(
```

Comment:
is this only being used in one place? I agree with Di's comments here - inlining smaller methods like this may improve readability and make the code easier to understand